### PR TITLE
MTDSA-7756 update dependencies

### PR DESCRIPTION
**Pull request by yangwong9664**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ object AppDependencies {
     ws,
     "uk.gov.hmrc" %% "bootstrap-backend-play-27" % "2.20.0",
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:25]: This should be 3.2.0

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ object AppDependencies {
...
     "uk.gov.hmrc"   %% "domain"            % "5.10.0-play-27",
-    "org.typelevel" %% "cats-core"         % "2.2.0",
+    "org.typelevel" %% "cats-core"         % "2.3.0",
     compilerPlugin("com.github.ghik" % "silencer-plugin" % "1.7.1" cross CrossVersion.full),
     "com.github.ghik" % "silencer-lib" % "1.7.1" % Provided cross CrossVersion.full,
     "com.chuusai"   %% "shapeless"         % "2.3.3"
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:30]: This should be 2.4.0-M1

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -33,8 +33,8 @@ object AppDependencies {
   def test(scope: String = "test, it"): Seq[sbt.ModuleID] = Seq(
     "org.scalatest"          %% "scalatest"          % "3.2.2"             % scope,
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:34]: scalatest should be bumped to 3.3.0-SNAP2. Also jettyFromWiremockVersion should be bumped to 9.4.35.v20201120. The one we are using currently is 2018 version.

**Review by Taiwo-AB — CHANGES_REQUESTED** [review]
Also, these plugins.sbt needs updating. These are the following versions.
uk.gov.hmrc -> sbt-auto-build -> 2.11.0
uk.gov.hmrc -> sbt-git-versioning -> 2.2.0
uk.gov.hmrc -> sbt-artifactory -> 1.10.0
uk.gov.hmrc -> sbt-distributables -> 2.1.0
com.typesafe.play -> sbt-plugin -> 2.7.5
org.scalastyle -> scalastyle-sbt-plugin -> 1.0.0
org.scoverage -> sbt-scoverage -> 1.6.1
com.timushev.sbt -> sbt-updates -> 0.5.1

File: project/AppDependencies.scala (hunk 3 above)
Comment (yangwong9664) [reply project/AppDependencies.scala:34]: It is better to stick to main releases, as they are generally safer. I'll keep Scalatest at 3.2.2

File: project/AppDependencies.scala (hunk 2 above)
Comment (yangwong9664) [reply project/AppDependencies.scala:30]: That release tag doesn't look like a main release, generally safer to stick to main releases/versions

File: project/AppDependencies.scala (hunk 3 above)
Comment (Taiwo-AB) [reply project/AppDependencies.scala:34]: Ok, but it still needs to be bumped to 3.2.3 as that's the latest main release

Comment (yangwong9664) [reply project/AppDependencies.scala:34]: If that's the case then upgrading to 3.2.3 is acceptable, as we don't typically upgrade to non-main releases

Comment (Taiwo-AB) [reply project/AppDependencies.scala:34]: yep no worries

Comment (BetaDraconis) [reply project/AppDependencies.scala]: https://github.com/hmrc/individuals-income-received-api/pull/208#discussion_r539203768

File: project/AppDependencies.scala (hunk 2 above)
Comment (Taiwo-AB) [reply project/AppDependencies.scala:30]: This is what we are using in most of other services so I think this one is safe to be bumped to that 2.4.0-M1. Example is losses in prod [here](https://github.com/hmrc/individual-losses-api/blob/master/project/AppDependencies.scala#L31)

Comment (yangwong9664) [reply project/AppDependencies.scala:30]: I don't think that's your call to make. They may have been upgraded by mistake. If we do go ahead with 2.4.0-M1, then it's better to get consensus from the team

Comment (Taiwo-AB) [reply project/AppDependencies.scala:30]: Not saying it is my call, just saying it is safe to do so as it is done for most of our services. If we do want to change, then I guess your other PRs which use 2.4.0-M1 currently and other services will need this changing

**Pull request merged**